02-feature-engineering.Rmd

```diff
@@ -984,39 +984,39 @@ Both are quite close, but you can see the natural spline does not dip quite as c
 
 #### Basis expansion with {recipes}
 
-Basis expansion via recipes proceeds in much the same way as we've seen before. We just specify a step specifying the the type of basis expansion we want to conduct, along with any additional arguments like the degrees of freedom (number of knots).
+Basis expansion via recipes proceeds in much the same way as we've seen before. We just specify a step identifying the the type of basis expansion we want to conduct, along with any additional arguments like the degrees of freedom (number of knots).
 
 Let's try building a dataset using basis expansion with a B-Spline
 
 ```{r }
-bs_d_rec <- recipe(y ~ x, data = sim) %>% 
+bs_d <- recipe(y ~ x, data = sim) %>% 
   step_bs(x, deg_free = 8) %>% 
   prep() %>% 
   bake(new_data = NULL)
 
-bs_d_rec
+bs_d
 ```
 
 And as you can see, we get similar output to polynomial regression, but we're just using a spline for our basis expansion now instead of polynomials. Note that the degrees of freedom can also be set to `tune()` and trained withing `tune::tune_grid()`. In other words, the amount of basis expansion (degree of "wiggliness") can be treated as a hyperparamter to be tuned.
 
 Once the basis expansion is conducted, we can model it just like any other linear model.
 
 ```{r }
-bs_mod_rec <- lm(y ~ x_bs_1 + x_bs_2 + x_bs_3 +
+bs_mod_2 <- lm(y ~ x_bs_1 + x_bs_2 + x_bs_3 +
                      x_bs_4 + x_bs_5 + x_bs_6 +
                      x_bs_7 + x_bs_8,
-                 data = bs_d_rec)
+                 data = bs_d)
```

Comment:
A note on this section -- I removed the `_rec` suffix from these objects because it might be confusing based on the precedent you've set by naming recipe objects with `_rec` (and these are not actually recipes themselves,; they are a dataframe and model object). I'm assuming the `_rec` suffix in this case was to specify that this is following a {recipes} approach -- so maybe we could instead name these with `_recipes` if you want to still make that extra clear? Let me know, and I'm happy to change it!

Reply:
Yes, that is confusing. Hmm.... not sure what's the best approach here. Why don't you just do what you think is best for now and we can always revisit.

Reply:
Sounds good, I'll just keep the changes I made for now